flex-checks/src/main/java/org/sonar/flex/checks/FunctionComplexityCheck.java

```diff
@@ -60,10 +59,10 @@ public void init() {
   @Override
   public void leaveNode(AstNode node) {
     SourceFunction function = (SourceFunction) getContext().peekSourceCode();
-    if (function.getInt(FlexMetric.COMPLEXITY) > maximumFunctionComplexityThreshold) {
-      getContext().createLineViolation(this,
-        "Function has a complexity of {0,number,integer} which is greater than {1,number,integer} authorized.", node,
-        function.getInt(FlexMetric.COMPLEXITY), maximumFunctionComplexityThreshold);
+    int functionComplexity = function.getInt(FlexMetric.COMPLEXITY);
+    if (functionComplexity > maximumFunctionComplexityThreshold) {
+      String message = String.format("Function has a complexity of %s which is greater than %s authorized.", functionComplexity, maximumFunctionComplexityThreshold);
+      createIssueWithCost(message, node, functionComplexity - maximumFunctionComplexityThreshold);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this substraction operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)